app/js/trouble.js

```diff
@@ -53,15 +55,15 @@
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
+				for (var i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this does remove the "making" of a function in a loop, what's desired here is to pass a reference to a function here, rather than invoking a function (which means it will be called as soon as the code executes rather than waiting for the timeout event to fire.) There's a couple of ways to accomplish this: declare the function outside the loop and pass the function name in as the first argument to setTimeout; assign the anonymous function to a  variable outside the loop and pass in the variable name.